### PR TITLE
Add `path` channel for sorting line's path + Replace `config.mark.sortLineBy`

### PR DESCRIPTION
**Pull request by kanitw**
Note: Documentation will be written in another huge doc writing PR.

### CODE REVIEW

**Comment by domoritz** [comment]
We should add a warning that path can only be used with line. Should be done in https://github.com/vega/vega-lite/issues/726

**Comment by kanitw** [comment]
Yeah it is a part of 

> all the specified encoding channels are supported by the mark type 

in #726

**Comment by kanitw** [comment]
I already added line to path's supported mark type in channel.ts, so that should be done automatically with 

> all the specified encoding channels are supported by the mark type

File: src/schema/encoding.schema.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ export interface Encoding {
   color?: FieldDef;
   size?: FieldDef;
   shape?: FieldDef;
+  path?: FieldDef;
```
Comment (domoritz) [inline src/schema/encoding.schema.ts]: | [FieldDef]?

Comment (kanitw) [inline src/schema/encoding.schema.ts]: DOne

**Comment by domoritz** [comment]
Fix conflicts

**Comment by kanitw** [comment]
Fixed

**Pull request merged**